### PR TITLE
Bugfix/sim 1615/gamma cache

### PR DESCRIPTION
**Pull request by danielsf**
The magnitude uncertainty calculations in PhotometryMixins.py were caching m5 and gamma parameter values in a haphazard way.  This pull request imposes a little more control on the association between the two parameters.

### CODE REVIEW

File: python/lsst/sims/catUtils/mixins/PhotometryMixin.py [hunk 1]
```diff
@@ -108,6 +71,16 @@ def _magnitudeUncertaintyGetter(self, column_names, bandpassDict_name):
         associated with the uncertainties calculated by this getter
         (the 'xx' in the 'sigma_xx' above)
 
+        @param [in] m5_names are the keys to the self.obs_metadata.m5 dict
+        corresponding to the bandpasses in column_names (e.g. in the case
+        of galaxies, the magntiude columns
```
Comment (jbkalmbach) [inline python/lsst/sims/catUtils/mixins/PhotometryMixin.py]: typo in magnitude

File: python/lsst/sims/catUtils/mixins/PhotometryMixin.py [hunk 2]
```diff
@@ -128,12 +101,29 @@ def _magnitudeUncertaintyGetter(self, column_names, bandpassDict_name):
...
-        return self.calculateMagnitudeUncertainty(magnitudes, getattr(self, bandpassDict_name),
-                                                  obs_metadata=self.obs_metadata)
+        try:
+            param_arr = numpy.array(
+                                    [[self.obs_metadata.m5[mname], self._gamma_cache[mname]]
+                                    if name in self._actually_calculated_columns
+                                    else [numpy.NaN, numpy.NaN]
+                                    for mname, name in zip(m5_names, column_names)]
+                                   ).transpose()
+        except KeyError as kk:
```
Comment (jbkalmbach) [inline python/lsst/sims/catUtils/mixins/PhotometryMixin.py]: Did you add a unit test for this error anywhere? I didn't see it.

Comment (danielsf) [inline python/lsst/sims/catUtils/mixins/PhotometryMixin.py]: No.  There is no test.  I will add one.

Comment (danielsf) [inline python/lsst/sims/catUtils/mixins/PhotometryMixin.py]: done

**Comment by jbkalmbach** [comment]
Good to merge.

**Pull request merged**